collect_app/src/main/java/org/odk/collect/android/backgroundwork/AutoSendTaskSpec.java

```diff
@@ -169,21 +167,9 @@ private boolean networkTypeMatchesAutoSendSetting(NetworkInfo currentNetworkInfo
     /**
      * Returns true if at least one form currently on the device specifies that all of its filled
      * forms should auto-send no matter the connection type.
-     * <p>
-     * TODO: figure out where this should live
      */
     private boolean atLeastOneFormSpecifiesAutoSend() {
-        FormsDao dao = new FormsDao();
-
-        try (Cursor cursor = dao.getFormsCursor()) {
-            List<Form> forms = dao.getFormsFromCursor(cursor);
-            for (Form form : forms) {
-                if (Boolean.valueOf(form.getAutoSend())) {
-                    return true;
-                }
-            }
-        }
-        return false;
+        return formsRepository.getAll().stream().anyMatch(form -> parseBoolean(form.getAutoSend()));
```

Comment:
This could be optimized but doesn't feel worth it given we generally assume the form list on device is pretty small.

Reply:
Agreed the structure is fine. How do you decide when to do the filtering on a list of all forms vs. introducing a new repository method (e.g. `getAllWithAutoSend` or maybe even `hasAnyWithAutoSend`)? I see value to centralizing and hiding away as much of the filtering as possible. I also understand the value of keeping a tighter API so maybe that's what won out here.

Reply:
Good question. I find forcing myself to write tests for any new method is a nice way to find that balance. If it's something that's only used in one place and really driven out by a higher level test it means I probably don't need a repo method for it but if it's common then or nuanced (sorting etc) then it's nicer to have a repo method I think. That's a pretty fuzzy answer but does it make sense?

Reply:
Yup! I'm somewhat surprised `getAllWithAutoSend` wouldn't get reused but it sounds like it was a deliberate choice and you do know you have coverage which are the things that matter.

Reply:
Yeah true. I'm honestly in two minds. We should potentially think about moving call sites to Kotlin, so we can start using default params. Then we'd just need a `getAllWith` method to cover most cases. 

For the moment I think this code is fine though. If we end up doing that operation twice I'd have no problem adding a method to the interface.